File: Lab7/t2.cpp
#include <stdio.h>
int main() {
	char inputString[50];
	int i;
	puts("Please input a lower-case English word: ");
	for (i = 0; i < 50; i++) {
		scanf("%c", &inputString[i]);
		if (inputString[i] == '\n' || inputString[i] == ' ')//break while it is not a word but a sentence
			break;
		if (!(inputString[i] <= 'z' && inputString[i] >= 'a')){//check if input is a lower-case English word
			puts("Please input a lower-case English word!");
			return -1;
		}
	}
	for (int j = 0; j < i / 2; j++) {//check if it is a palindrome
		if (inputString[j] != inputString[i - 1 - j]) {
			puts("It's not a palindrome.");
			return 0;
		}
	}
	puts("It's a palindrome.");
	return 0;
}

File: Lab7/t2.java
import java.util.Scanner;
public class t2{
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        String inputWord;
        System.out.println("Please input a lower-case English word: ");
        inputWord = scan.next();
        scan.close();
        for (int i = 0; i < inputWord.length(); i++){
            if (!(inputWord.charAt(i) <= 'z' && inputWord.charAt(i) >= 'a')){
                System.out.println("Please input a lower-case English word!");
                System.exit(-1);
            }
        }
        for (int i = 0; i < inputWord.length() / 2; i++){
            if (inputWord.charAt(i) != inputWord.charAt(inputWord.length() - 1 - i)){
                System.out.println("It's not a palindrome.");
                System.exit(0);
            }
        }
        System.out.println("It's a palindrome.");
    }
}